experimental/dds/tree2/src/feature-libraries/sequence-field/sequenceFieldEditor.ts

```diff
@@ -91,21 +94,59 @@ export const sequenceFieldEditor = {
 		count: number,
 		destIndex: number,
 		id: ChangesetLocalId,
-	): [moveOut: Changeset<never>, moveIn: Changeset<never>] {
+	): Changeset<never> {
+		const moveIn: Mark<never> = {
+			type: "MoveIn",
+			id,
+			count,
+			cellId: { localId: id },
+		};
 		const moveOut: Mark<never> = {
 			type: "MoveOut",
 			id,
 			count,
 		};
+		const indexAfterMoveOut = sourceIndex + count;
+		const marks = new MarkListFactory<never>();
+		marks.pushOffset(Math.min(sourceIndex, destIndex));
+		if (destIndex <= sourceIndex) {
+			// The destination is fully before the source
+			marks.pushContent(moveIn);
+			marks.pushOffset(sourceIndex - destIndex);
+			marks.pushContent(moveOut);
+		} else if (indexAfterMoveOut <= destIndex) {
+			// The destination is fully after the source
+			marks.pushContent(moveOut);
+			marks.pushOffset(destIndex - indexAfterMoveOut);
+			marks.pushContent(moveIn);
+		} else {
+			const count1 = destIndex - sourceIndex;
+			const count2 = indexAfterMoveOut - destIndex;
+			// The destination is in the middle of the source
+			marks.pushContent({ ...moveOut, count: count1 });
+			marks.pushContent(moveIn);
+			marks.pushContent({ ...moveOut, count: count2, id: brand<MoveId>(id + count1) });
+		}
+		return marks.list;
+	},
+
+	moveOut(sourceIndex: number, count: number, id: ChangesetLocalId): Changeset<never> {
+		const moveOut: Mark<never> = {
+			type: "MoveOut",
+			id,
+			count,
+		};
+		return markAtIndex(sourceIndex, moveOut);
+	},
 
+	moveIn(destIndex: number, count: number, id: ChangesetLocalId): Changeset<never> {
 		const moveIn: Mark<never> = {
 			type: "MoveIn",
 			id,
 			count,
 			cellId: { localId: id },
 		};
-
-		return [markAtIndex(sourceIndex, moveOut), markAtIndex(destIndex, moveIn)];
+		return markAtIndex(destIndex, moveIn);
 	},
 
 	return(
```

Comment:
It seems like we should either delete the `return` builder or update it to use the same scheme as move.